### PR TITLE
UpdateTrack enum removed from AppCenterSettings

### PR DESCRIPTION
**Pull request by olmisa**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

'UpdateTrack' enum was removed from the 'AppCenterSettings' class. A new attribute added to draw dropdown in the editor GUI.

## Related PRs or issues

[AB#78905](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/78905)

### CODE REVIEW

File: Assets/AppCenter/Editor/CustomDropDownPropertyDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+[CustomPropertyDrawer(typeof(CustomDropDownPropertyAttribute))]
+public class CustomDropDownPropertyDrawer : PropertyDrawer
+{
+    bool initialized = false;
```
Comment (guperrot) [inline Assets/AppCenter/Editor/CustomDropDownPropertyDrawer.cs]: Those private fields should use `_` prefix.

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using UnityEngine;
+
+[AttributeUsage(AttributeTargets.Field, AllowMultiple = true)]
+public class CustomDropDownPropertyAttribute : PropertyAttribute
+{
+    public CustomDropDownPropertyAttribute(string _key, int _value)
```
Comment (guperrot) [inline Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Remove `_`

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        SelectedKey = _key;
+        SelectedValue = _value;
+    }
+
+    public CustomDropDownPropertyAttribute()
+    { }
+
+    public string SelectedKey;
```
Comment (guperrot) [inline Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Can this be properties?

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class CustomDropDownPropertyAttribute : PropertyAttribute
+{
+    public CustomDropDownPropertyAttribute(string _key, int _value)
+    {
+        SelectedKey = _key;
+        SelectedValue = _value;
+    }
+
+    public CustomDropDownPropertyAttribute()
+    { }
```
Comment (guperrot) [inline Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Is this formatted correctly?

File: Assets/AppCenter/Editor/CustomDropDownPropertyDrawer.cs (hunk 1 above)
Comment (olmisa) [reply Assets/AppCenter/Editor/CustomDropDownPropertyDrawer.cs]: Fixed

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs (hunk 2 above)
Comment (olmisa) [reply Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Fixed

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs (hunk 3 above)
Comment (olmisa) [reply Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Fixed

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs (hunk 4 above)
Comment (olmisa) [reply Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs]: Fixed

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using UnityEngine;
+
+[AttributeUsage(AttributeTargets.Field, AllowMultiple = true)]
```
Comment (annakocheshkova) [inline Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs:7]: AllowMultiple?

File: Assets/AppCenter/AppCenterSettings.cs [hunk 6]
```diff
@@ -48,7 +47,9 @@ public class AppCenterSettings : ScriptableObject
 
     public LogLevel InitialLogLevel = LogLevel.Info;
 
-    public UpdateTrack UpdateTrack = UpdateTrack.Public;
+    [CustomDropDownProperty("Public", 1)]
```
Comment (annakocheshkova) [inline Assets/AppCenter/AppCenterSettings.cs:50]: Don't we need to start the count from 0?

File: Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs (hunk 5 above)
Comment (MatkovIvan) [reply Assets/AppCenter/Properties/CustomDropDownPropertyAttribute.cs:7]: Yes, there are 2 attribute applied to property

**Review by MatkovIvan — APPROVED** [review]

File: Assets/AppCenter/AppCenterSettings.cs (hunk 6 above)
Comment (MatkovIvan) [reply Assets/AppCenter/AppCenterSettings.cs:50]: I guess it's just mapping to underlying API

Comment (olmisa) [reply Assets/AppCenter/AppCenterSettings.cs:50]: I want to use here 'UpdateTrack' values to use them later and reduce manipulations like '+-1'.

**Review by annakocheshkova — APPROVED** [review]
Tested, LG

**Review by guperrot — APPROVED** [review]

**Pull request merged**